File: Data Structures & Algorithms/three-integer-sum/submission-1.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> sol;
        sort(nums.begin(), nums.end());
        for (int k = 0; k < nums.size() - 2; k++) {
            int i = k+1, j = nums.size() - 1;
            while (i < j) {
                if (nums[i] + nums[j] > -nums[k]) j--;
                else if (nums[i] + nums[j] < -nums[k]) i++;
                else {
                    sol.push_back({nums[k],nums[i],nums[j]});
                    i++; j--;
                };
            }
        } 
        set<vector<int>> us(sol.begin(), sol.end());
        vector<vector<int>> ans(us.begin(), us.end());
        return ans;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}
        heap = []
        sol = []
        for x in nums:
            freq[x] = freq.get(x,0)-1
        for x in freq:
            heapq.heappush(heap,(freq[x],x))
        for i in range(k):
            val = heapq.heappop(heap)
            sol.append(val[1])
        return sol